### PR TITLE
fix vbe drawing circle speed

### PR DESCRIPTION
**Pull request by terminal-cs**
still cant merge because stupid appveryor cant build cosmos because dotnet 5 still isn't finished :/

### CODE REVIEW

**Review by valentinbreiz — APPROVED** [review]

File: source/Cosmos.System2/Graphics/VBECanvas.cs [hunk 1]
```diff
@@ -369,6 +405,93 @@ public override void DrawImage(Image aImage, int aX, int aY)
...
+        /// <exception cref="Exception">Thrown on memory access violation.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown on fatal error.</exception>
+        public override void DrawImageAlpha(Image image, int x, int y)
+        {
+            Global.mDebugger.SendInternal($"Drawing image of size {image.Width}x{image.Height} array size {image.rawData.Length} with alpha chanel...");
+            for (int _x = 0; _x < image.Width; _x++)
+            {
+                for (int _y = 0; _y < image.Height; _y++)
+                {
+                    _VBEDriver.SetVRAM((uint)GetPointOffset(y + _y, x + _x), (byte)image.rawData[_x + _y * image.Width]);
```
Comment (quajak) [inline source/Cosmos.System2/Graphics/VBECanvas.cs:423]: Isnt this ignoring the alpha data of the image? Isnt the implementation the same as in the base case.

File: source/Cosmos.System2/Graphics/VBECanvas.cs [hunk 2]
```diff
@@ -369,6 +405,93 @@ public override void DrawImage(Image aImage, int aX, int aY)
...
+
+        /// <summary>
+        /// Draw string.
+        /// </summary>
+        /// <param name="str">string to draw.</param>
+        /// <param name="aFont">Font used.</param>
+        /// <param name="pen">Color.</param>
+        /// <param name="x">X coordinate.</param>
+        /// <param name="y">Y coordinate.</param>
+        public override void DrawString(string str, Font aFont, Pen pen, int x, int y)
```
Comment (quajak) [inline source/Cosmos.System2/Graphics/VBECanvas.cs:449]: Why is this method overriden?

File: source/Cosmos.System2/Graphics/VBECanvas.cs [hunk 3]
```diff
@@ -369,6 +405,93 @@ public override void DrawImage(Image aImage, int aX, int aY)
...
+        }
+
+        /// <summary>
+        /// Draw string.
+        /// </summary>
+        /// <param name="str">char to draw.</param>
+        /// <param name="aFont">Font used.</param>
+        /// <param name="pen">Color.</param>
+        /// <param name="point">Point of the top left corner of the char.</param>
+        public void DrawChar(char c, Font aFont, Pen pen, Point point)
```
Comment (quajak) [inline source/Cosmos.System2/Graphics/VBECanvas.cs:465]: Same for this one

File: source/Cosmos.System2/Graphics/VBECanvas.cs [hunk 4]
```diff
@@ -369,6 +405,93 @@ public override void DrawImage(Image aImage, int aX, int aY)
...
+
+        /// <summary>
+        /// Draw char.
+        /// </summary>
+        /// <param name="str">char to draw.</param>
+        /// <param name="aFont">Font used.</param>
+        /// <param name="pen">Color.</param>
+        /// <param name="x">X coordinate.</param>
+        /// <param name="y">Y coordinate.</param>
+        public void DrawChar(char c, Font aFont, Pen pen, int x, int y)
```
Comment (quajak) [inline source/Cosmos.System2/Graphics/VBECanvas.cs:478]: Shouldnt this method be an override?

**Comment by terminal-cs** [comment]
closing because i do not have this branch anymore so i cannot make changes.

**Pull request closed without merging**